nistats/reporting/glm_reporter.py

```diff
@@ -0,0 +1,561 @@
+import io
+import string
+import os
+
+from collections import OrderedDict
+
+import pandas as pd
+
+from matplotlib import pyplot as plt
+from nilearn.datasets import load_mni152_template
+from nilearn.plotting import (plot_glass_brain,
+                              plot_roi,
+                              plot_stat_map,
+                              )
+from nilearn.plotting.js_plotting_utils import HTMLDocument
+
+from nistats.reporting import (plot_contrast_matrix,
+                               plot_design_matrix,
+                               get_clusters_table,
+                               )
+from nistats.thresholding import map_threshold
+
+
+html_template_root_path = os.path.dirname(__file__)
+
+
+def make_glm_report(
+        model,
+        contrasts,
+        title='auto',
+        roi_img=None,
+        bg_img='MNI 152 Template',
+        threshold=3.09,
+        alpha=0.01,
+        cluster_threshold=None,
+        height_control='fpr',
+        min_distance=8.,
+        plot_type='stat',
+        display_mode=None,
+        cut_coords=None,
+        nb_width=1600,
+        nb_height=800,
+        ):
+    """ Returns HTMLDocument object for a report which shows
+    all important aspects of a fitted GLM.
+    The object can be opened in a browser or saved to disk.
+    
+    Usage:
+        report = make_glm_report(model, contrasts)
+        report.open_in_browser()
+        report.save_as_html(destination_path)
+    
+    Parameters
+    ----------
+    model: FirstLevelModel or SecondLevelModel object
+        A fitted first or second level model object.
+        
+    contrasts: Dict[string, ndarray] , String
+    
+    title: str, default 'auto'
+        Text to represnt the web page's title and primary heading.
+        
+    threshold: float
+        Cluster forming threshold in same scale as `stat_img` (either a
+        p-value or z-scale value).
+        Default is 3.09
+        
+    alpha: float
+        P- value for clustering.
+
+    cluster_threshold : int or None, optional
+        Cluster size threshold, in voxels.
+        
+    height_control: str
+
+    min_distance: float, optional
+        Minimum distance between subpeaks in mm. Default is 8 mm.
+        
+    bg_img: Nifti image
+        Default is the MNI152 template
+        
+    display_mode: String, optional
+        Default is 'z' if plot_type is 'stat'; 'ortho' if plot_type is 'glass'.
+        
+    plot_type: String. ['stat' (default)| 'glass']
+        Specifies the type of plot to be drawn for the statistical maps.
+        
+    Returns
+    -------
+    report_text: HTMLDocument Object
+        Contains the HTML code for the GLM Report.
+    """
+    if not display_mode:
+        if plot_type == 'stat':
+            display_mode = 'z'
+        elif plot_type == 'glass':
+            display_mode = 'lzry'
+    cluster_threshold = cluster_threshold if cluster_threshold else 0
```

Comment:
I'll do that.
I had set the default size to 1 since 1 voxel is the minimum possible cluster size, we can't have 0 voxels in a cluster, but elsewhere, don't remember where I saw minimum as 0 so I changed this but I think the minimum should be 1 not 0. WDYT?

Reply:
It depends on how you handle equality cases !
i'd rather have 0 be default, assuming that stict inequalities are enforced afterwards. The problem with non-zero values is that you further need to figure out in which units things are specified (voxels, mm^3, etc.)